server/src/main/java/org/elasticsearch/DistributedArchitectureGuide.md

```diff
@@ -0,0 +1,281 @@
+# Distributed Area Team Internals
```

Comment:
I think it is better to move those files closer to the root, namely in `server` directory since they are not a java source files

Reply:
Sounds good, done. There isn't a /docs/internal/ directory, so I presume you meant create one.